packages/core/test/unit/lib/markdown-it/highlight/HightlightRule.test.ts

```diff
@@ -0,0 +1,83 @@
+import { HighlightRule, HIGHLIGHT_TYPES } from '../../../../../src/lib/markdown-it/highlight/HighlightRule';
+import { HighlightRuleComponent } from '../../../../../src/lib/markdown-it/highlight/HighlightRuleComponent';
+
+describe('parseAllRules', () => {
+  test('parses multiple rules correctly', () => {
+    const allRules = '3,4-5';
+    const rules = HighlightRule.parseAllRules(allRules, 0, 'line1\nline2\nline3\nline4\nline5');
+    expect(rules).toHaveLength(2);
+    expect(rules[0]).toBeInstanceOf(HighlightRule);
+    expect(rules[1]).toBeInstanceOf(HighlightRule);
+  });
+
+  test('ignores invalid rules', () => {
+    const allRules = 'invalid,3-4';
+    const rules = HighlightRule.parseAllRules(allRules, 0, 'line1\nline2\nline3\nline4');
+    expect(rules).toHaveLength(1);
```

Comment:
Same here, could we assert the rule generated is actually 3-4

Reply:
Done